src/amqp.js

```diff
@@ -0,0 +1,344 @@
+      if (channelContainsMessages(channelInfo)) {
+        debug(`Prepared to getOne from queue: ${queue}`);
+        // Returns false if 0 items in queue
+        const message = await channel.get(queue);
+
+        // debugData(`Message: ${inspect(message, {colors: true, maxArrayLength: 3, depth: 3})}`);
+        // Do not spam the logs
+
+        if (message) {
+          debugDev(`Got one from queue: ${queue}`);
+          const headers = getHeaderInfo(message);
+          const records = transformMessagesToRecords([message]);
+          return {headers, records, messages: [message]};
+        }
+
+        return false;
+      }
+
+      return false;
+    } catch (error) {
+      handleAmqpErrors(error);
+    }
+  }
+
+  // MARK: Ack messages
+  /**
+   * Ack messages - Acknowledge the given messages
+   *
+   * @param {Array[Object]} messages Array of message objects
+   * @returns {void} returns nothing
+   */
+  async function ackMessages(messages) {
+    messages.forEach(message => {
+      debugDev(`Ack message ${message.properties.correlationId}`);
+      channel.ack(message);
+    });
+    await setTimeoutPromise(10);
+    return;
+  }
+
+  // MARK: Nack messages
+  /**
+   * Nack messages - Reject a messages back to queue
+   *
+   * @param {Array[Object]} messages Array of message objects
+   * @returns {void} returns nothing
+   */
+  async function nackMessages(messages) {
+    messages.forEach(message => {
+      debugDev(`Nack message ${message.properties.correlationId}`);
+      channel.nack(message, false, true);
+    });
+    await setTimeoutPromise(10);
+    return;
+  }
+
+  // MARK: Send to queue
+  /**
+   * Sends to queue
+   * Send to queue uses blob id and status to form queue id.
+   * E.g. 905e283f-2644-4d26-8432-6634fbbc0161.PENDING_LOOKUPS
+   *
+   * @param {Object} params object: blobId, status, data
+   * @param {uuid} params.blobId BlobId for blob to be handled
+   * @param {string} params.status Status for queue
+   * @param {object} params.data Data object sent to queue
+   * @returns {boolean} returns true on success
+   */
+  async function sendToQueue({blobId, status, data}) {
+    debug(`sendToQueue`);
+    // eslint-disable-next-line no-useless-catch
+    try {
+      const {queue} = await generateQueueId({blobId, status});
+      debug(`Queue ${queue}`);
+      debugDev(`Queue Asserted: ${queue}`);
+
+      debugData(`Data ${JSON.stringify(data)}`);
+
+      debugDev(`Actually sendToQueue: ${queue}`);
+      await channel.sendToQueue(
+        queue,
+        Buffer.from(JSON.stringify({data})),
+        {
+          correlationId: blobId,
+          persistent: true
+        }
+      );
+      debugDev(`Send message to queue: ${queue}`);
+
+      return true;
+    } catch (error) {
+      debug(`SendToQueue errored: ${JSON.stringify(error)}`);
+      handleAmqpErrors(error);
+    }
+  }
+
+  async function removeQueue({blobId, status}, allowMessageLoss = false) {
```

Comment:
Missing comments

Reply:
// MARK: Remove queue
  /**
   * Remove queue uses blob id and status to form queue id.
   * E.g. 905e283f-2644-4d26-8432-6634fbbc0161.PENDING_LOOKUPS
   *
   * @param {Object} params object: blobId, status
   * @param {uuid} params.blobId BlobId for blob to be handled
   * @param {string} params.status Status for queue
   * @param {boolean} allowMessageLoss Allow removing a queue that has messages
   */

Reply:
Double code. Merging it so delete queue function